### DISCUSSION TITLE
MIMXRT105x/106x support

### DISCUSSION THREAD

**Discussion by cakeslob** [mbed-ce/mbed-os#109] [Q&A]
"Basically, the MIMXRT1051 is the base model, and code compatible with it will run on any of the other chips. The other part numbers simply add features on top of that."

Reading this,  does this mean the MIMXRT1062 aka Teensy4/4.1  , is "supported" by mbed-os and has been this whole time? 

Also, good work with the community edition. I have high hopes for the project, as it made migrating from various chips during the shortage relatively easy , despite my lack of programming knowledge.

**Comment by multiplemonomials** [comment]
That was exactly what I was getting at!  I am fairly sure that the Teensy could be supported with relatively minor changes.  Basically, the memory configuration on that board is substantially different from the MIMXRT1050_EVK, and there would be some mods needed to the bootup code and linker script.  I'm pretty sure I know how to make these mods thanks to some experiences at my job.

Additionally, we would need to get Mbed to generate a hex file in the correct format for Teensy Loader so that you could flash the board.  I'm not 100% sure what it needs.

Do you have a Teensy 4.x available to test with?  If so, I could try and make a branch for it.

**Comment by cakeslob** [comment]
Ok, so it wasnt flat supported but undocumented, it needs a little work.

"Do you have a Teensy 4.x available to test with? If so, I could try and make a branch for it." 

Yeah, I have a 4.0 I could test with, 

but full disclosure Im not a computer programmer

**Reply by multiplemonomials** [reply]
Okay!  I took an initial crack at implementing this https://github.com/mbed-ce/mbed-os/pull/113 .  Please check out and test the `dev/teensy-4.0-support` branch when you can!

If you build for the TEENSY_40` target, it should theoretically generate a hex file which can be flashed with Teensy Loader.  Console output should work, the board should show up a s a USB serial device and things that you print should show up.  Note that you may run into some issues with peripherals not being mapped, so right now I'm just interested in if you can get console output and if you can flash the led.

**Reply by multiplemonomials** [reply]
Did you have a chance to try it out?

**Reply by cakeslob** [reply]
Sorry,not yet. There is a learning curve I must overcome with vscode coming from an mbed studio environment. If you have a hello world hex file you would like me to load, that is the fastest way.

**Reply by multiplemonomials** [reply]
Here ya go!  Sorry for delay, was on vacation.  
[mbed-ce-hello-world-teensy40.zip](https://github.com/mbed-ce/mbed-os/files/10328311/mbed-ce-hello-world-teensy40.zip)

**Reply by JohnK1987** [reply]
@cakeslob did you have space for the test?

**Reply by multiplemonomials** [reply]
I made a general call for help: https://forum.pjrc.com/threads/72107-Testers-Needed-Mbed-OS-Port-for-Teensy-4-0?p=320208#post320208

**Comment by JojoS62** [comment]
I have an older Teensy, should be 3.6. Is it also supported? I never had a closer look to this board.

**Reply by multiplemonomials** [reply]
That uses a Freescale MK20, so it's not affected by my MR
